Step4 구현

joont92 opened the pull request:
- QnAServiceTest의 setUp을 제거하고 각자 스텁을 쓰다가, 보기가 불편해서 각각 함수로 분리해봤습니다
- DeleteHistory는 어떻게 테스트해야할지 감이 잘 안오네용..

리뷰 부탁드립니다!!

joont92 committed:
test : QnAServiceTest setUp() 삭제

joont92 committed:
test : QnAServiceTest에서 stub 부분을 전부 메서드 분리

- answerRepository 사용하는 부분 lazy, persist 사용하도록 변경
	- addAnswer 추가
- QnAServiceTest에서 stub 부분을 전부 메서드 분리

joont92 committed:
feat : 질문 삭제 기능 구현

joont92 committed:
feat : deleteHistory 구현

phs1116 commented on src/main/java/nextstep/domain/Question.java:
```diff
@@ -51,12 +56,21 @@ public void update(User writer, QuestionBody newQuestionBody) {
         deleted = true;
+        histories.add(new DeleteHistory(ContentType.QUESTION, getId(), writer, LocalDateTime.now()));
+
+        getAnswers().stream()
```
answers를 일급콜렉션으로 만들어보면 어떨까요?

joont92 replied:
추가하였습니다

phs1116 commented on src/test/java/nextstep/service/QnAServiceTest.java:
```diff
@@ -84,38 +76,73 @@ public void setUp() {
+        답변이_포함된_질문_1건을_조회_가능하게함(1L);
 
-        softly.assertThat(question.isDeleted()).isTrue();
+        List<DeleteHistory> histories = qnaService.deleteQuestion(1L, writer);
```
DeleteHistory를 반환하도록 하신게 테스트때문이신가요?
mockito에서 verify 검색해보시고 해당 기능으로 검증도 해보면 어떨까요? :)

joont92 replied:
verify로 변경했습니다!

phs1116 reviewed (changes requested):
안녕하세요. 리뷰어 박현식입니다. :) 
테스트 작성 👍 
피드백 몇가지 추가했어요. 참고 부탁드려요 :)
궁금하신점 있으시면 언제든지 연락주세요!

joont92 committed:
fix : 피드백 적용

- QnAService 에서 DeleteHistory 반환하는 부분 삭제, verify 행위검증
- List<Answer> 일급 컬렉션 생성
- DeleteHistoryService 제거

joont92 commented:
DeleteHistory와 List<Answer> 일급컬렉션으로 구현하는 부분들이 조금 어렵네요. 잘 접해보지 못해서인것 같습니다.
감사합니다!

javajigi reviewed (approved):
피드백 반영 잘 했네요. 💯 
qna 미션까지 진행하느라 수고하셨어요.
